nectar/downloaders/threaded.py

```diff
@@ -257,7 +256,9 @@ def _fetch(self, request, session):
             )
             report.download_skipped()
 
-        except requests.ConnectionError:
+        except requests.ConnectionError, e:
+            if isinstance(e.strerror, httplib.BadStatusLine):
+                raise RetryError()
```

Comment:
Consider adding the url to the exception information. It's caught by the outer loop but it would be a better exception with more info about what went wrong.